docs/app/docs/configuration.md

```diff
@@ -24,6 +25,23 @@ This is a list of Nix packages that should be installed in your Devbox shell and
 
 You can add packages to your devbox.json using `devbox add <package_name>`, and remove them using `devbox rm <package_name>`
 
+### Env
+
+This is a a map of key-value pairs that should be set as Environment Variables when activating `devbox shell`, running a script with `devbox run`, or starting a service. These variables will only be set in your Devbox shell, and will have precedence over any environment variables set in your local machine or by [Devbox Plugins](guides/plugins.md).
+
+For example, you could set variable `$FOO` to `bar` by adding the following to your `devbox.json`:
+
+```json
+{
+    "env": {
+        "FOO": "bar"
+    }
+}
+```
+
+Currently, you can only set values using string literals, `$PWD`, and `$PATH`. Any other values will environment variables will not be expanded when starting your shell. 
```

Comment:
edit needed: Any other values ~will~ with environment variables will not be expanded when starting your shell.